File: src/cli/document/usage.ts
import type { Command } from "commander";

const USAGE_TEXT = `lin document — Document operations (search, list, get, create, update)

SEARCH:
  document search <text>                    Full-text search
    [--include-comments] [--include-archived]

LIST:
  document list                             List documents
    [--project <p>] [--creator <u>] [--include-archived]

GET:
  document get <id>                         Full details + content (markdown)
    Returns: title, content, url, icon, color, project, creator, updatedBy, timestamps

CREATE:
  document new <title>                      Create document
    [--project <p>] [--content <md>] [--icon <emoji>] [--color <hex>]

UPDATE (each is a subcommand):
  document update title <id> <new-title>
  document update content <id> <markdown>
  document update project <id> <project>    Move to project

IDS: UUID or slug ID (shown as "slugId" in output). "get" accepts either format.
PROJECT: Accepts UUID, slug ID, or name (case-insensitive).
CREATOR: Filter by user ID, name, display name, or email.
CONTENT: Markdown format. Full content returned by "get"; truncated in list/search.
PAGINATION: --limit <n> --cursor <token> on search and list.
`;

export function registerUsage(document: Command): void {
  document
    .command("usage")
    .description("Print detailed document command documentation (LLM-optimized)")
    .action(() => {
      console.log(USAGE_TEXT.trim());
    });
}

File: internal/cli/initiative/usage.go
package initiative

const usageText = `lin initiative — Manage Linear initiatives (replaces roadmap)

SEARCH & LIST:
  initiative search <text>                Search initiatives by name
    [--limit] [--cursor]
  initiative list                         List all initiatives
    [--status planned|active|completed]
    [--limit] [--cursor]
    Returns per item: id, slugId, url, name, status, health, targetDate, owner

GET:
  initiative get <id>            Initiative summary: id, slugId, url, name,
                                 description, content, status, health, targetDate,
                                 startedAt, completedAt, createdAt, updatedAt,
                                 owner{id,name}, creator{id,name}
  initiative projects <id>       Projects linked to an initiative
    [--limit] [--cursor]
    Returns per item: id, slugId, url, name, status, progress,
    lead, startDate, targetDate

CREATE:
  initiative new <name>          Create initiative
    [--description] [--owner] [--status] [--target-date]
    [--content] [--color] [--icon]

UPDATE:
  initiative update name <id> <value>
  initiative update description <id> <value>
  initiative update content <id> <value>
  initiative update status <id> <value>        planned|active|completed
  initiative update owner <id> <user>          name, email, or user ID
  initiative update target-date <id> <date>    YYYY-MM-DD
  initiative update color <id> <color>
  initiative update icon <id> <icon>

LIFECYCLE:
  initiative archive <id>        Archive initiative
  initiative unarchive <id>      Restore archived initiative
  initiative delete <id>         Permanently delete initiative

IDS: <id> accepts UUID, slug ID, or initiative name.
PAGINATION: --limit <n> --cursor <token> on list and projects.
STATUS: planned | active | completed`
